### ISSUE TITLE
Make use of inlining

### ISSUE THREAD

**Issue by Okamoi** [massiveinteractive/msignal#9]
Hi,

Thanks for that simple and efficient library. Would you mind inlining all the appropriate functions ?

I would do it myself and send a pull request, but I'm afraid I may be a little heavy handed with inlining. Might as well be done by someone who fully knows how it's best used :)

Thanks either way !

**Comment by thomasuster** [comment]
Hey @Okamoi . Inlining can be weird. I think if you inline you have to allocate more onto the stack, so I only add it if there's a very specific bottle-neck. Are you hitting a performance problem somewhere?

**Comment by Okamoi** [comment]
Not really. I just consistently see speed improvements on most code I inline with Haxe. I did notice that many good programmers use inline more sparingly than I do. I don't exactly understand why. Perhaps it's with good reason, or perhaps they come from languages with a compiler that inlines less cleverly than Haxe's.

I trust you know what's best better than I do so feel free to close the issue if you want :)
